docs/src/main/asciidoc/datasource.adoc

```diff
@@ -559,44 +549,28 @@ XA transactions on reactive datasources are not supported at the moment.
+If your transaction involves other non-datasource resources, keep in mind *those* resources might not support XA transactions, or might require additional configuration.
 ====
 
 If XA cannot be enabled for one of your datasources:
 
-* Be aware that enabling XA for all datasources _except one_ (and only one) is still supported
-through https://www.narayana.io/docs/project/index.html#d5e857[Last Resource Commit Optimization (LRCO)].
-* If you do not need a rollback for one datasource to trigger a rollback for other datasources,
-consider splitting your code into multiple transactions.
-To that end, use xref:transaction.adoc#programmatic-approach[`QuarkusTransaction.requiringNew()`]/xref:transaction.adoc#declarative-approach[`@Transactional(REQUIRES_NEW)`] (preferably)
-or xref:transaction.adoc#legacy-api-approach[`UserTransaction`] (for more complex use cases).
+* Be aware that enabling XA for all datasources _except one_ (and only one) is still supported through https://www.narayana.io/docs/project/index.html#d5e857[Last Resource Commit Optimization (LRCO)].
+* If you do not need a rollback for one datasource to trigger a rollback for other datasources, consider splitting your code into multiple transactions.
+To that end, use xref:transaction.adoc#programmatic-approach[`QuarkusTransaction.requiringNew()`]/xref:transaction.adoc#declarative-approach[`@Transactional(REQUIRES_NEW)`] (preferably) or xref:transaction.adoc#legacy-api-approach[`UserTransaction`] (for more complex use cases).
 
 [CAUTION]
 ====
-As a last resort, and for compatibility with Quarkus 3.8 and earlier,
-you may allow unsafe transaction handling across multiple non-XA datasources
-by setting `quarkus.transaction-manager.unsafe-multiple-last-resources` to `allow`.
-
-With this property set to `allow`, a transaction rollback
-could possibly be applied to only some of the non-XA datasources,
-with other non-XA datasources having already committed their changes,
-leaving your overall system in an inconsistent state.
-
-Alternatively, you can allow the same unsafe behavior,
-but with warnings when it is taken advantage of:
-
-* setting the property to `warn-each`
-would result in logging a warning on *each* offending transaction.
-* setting the property to `warn-first`
-would result in logging a warning on the *first* offending transaction.
-
-We do not recommend using this configuration property,
-and we plan to remove it in the future,
-so you should plan fixing your application accordingly.
-If you think your use case of this feature is valid and this option should be kept around,
-open an issue in the https://github.com/quarkusio/quarkus/issues/new?assignees=&labels=kind%2Fenhancement&projects=&template=feature_request.yml[Quarkus tracker]
+As a last resort, and for compatibility with Quarkus 3.8 and earlier, you might allow unsafe transaction handling across multiple non-XA datasources by setting `quarkus.transaction-manager.unsafe-multiple-last-resources` to `allow`.
```

Comment:
What does the "You might allow" imply here?  That you should do it? That you are allowed to do it? Either way, the wording should be replaced with something way more actionable.

Reply:
This should definitely be "may", not "might"

Reply:
What about this?

Only when all other options are exhausted, and for compatibility with Quarkus 3.8 and earlier, you can allow unsafe transaction handling across multiple non-XA datasources by setting `quarkus.transaction-manager.unsafe-multiple-last-resources` to `allow`.

Reply:
option 2  also looks good to me.

Reply:
Great, applying! Thank you!